Source/DafnyCore/Compilers/Compiler-java.cs

```diff
@@ -2136,15 +2187,18 @@ private class GenericArrayElementLvalue : ILvalue {
             throw new cce.UnreachableException();
         }
       } else {
-        // TODO-RS: This doesn't handle strings printed out as part of datatypes
-        bool isString = arg.Type.AsSeqType != null &&
-                        arg.Type.AsSeqType.Arg.IsCharType;
         bool isGeneric = arg.Type.AsSeqType != null &&
                          arg.Type.AsSeqType.Arg.IsTypeParameter;
-        if (isString) {
+        if (arg.Type.IsStringType) {
           TrParenExpr(arg, wr, false, wStmts);
           wr.Write(".verbatimString()");
-        } else if (isGeneric) {
+        } else if (arg.Type.IsCharType && UnicodeChars) {
+          wr.Write($"{DafnyHelpersClass}.ToCharLiteral(");
+          TrExpr(arg, wr, false, wStmts);
+          wr.Write(")");
+        } else if (isGeneric && !UnicodeChars) {
+          // This happens to not work when --unicode-char is true anyway,
```

Comment:
What do you mean "to not work"? Is the code buggy?

Reply:
Out of date comment, will remove.